### DISCUSSION TITLE
Download file issue

### DISCUSSION THREAD

**Discussion by onght** [Ylianst/MeshCentral#6242] [Issues]
I have Meshcentral running in Windows. When I tried to download file from a agent connected client, I will get an error "Couldn't download - Network issue" in Microsoft Edge. In Chrome and Brave browser, I will get and error with "Check Internet Connection". However, when I turn on "Inspect" in my browser, the download would be successful. I would be able to download a few files as long as I keep the "Inspect" windows in the browsers turned on and it will failed to download again. I will need to switch off the "Inspect" windows and switch it on again and it will work again. What could be the problem?

**Comment by onght** [comment]
I just found the issue is related to browser and the self signed certificates. Chrome, Brave and Edge block the transfer because it doesn't trust the self signed cert and reject the transfer. When I use Firefox, I don't have this issue. However, it is interesting to know what whenever I enabled the "Inspect" mode on the browser, the download is enable to function?

**Comment by silversword411** [comment]
Option 1: Use valid SSL on your mesh
Option 2: Open issue with chromium project to discuss their security defaults about downloads without full chain certs
